### PR TITLE
Adds tests for application creation

### PR DESCRIPTION
**Pull request by eugenesk24**
### Description
Adds tests for application analytics create page

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ps48 — APPROVED** [review]

**Review by kavithacm — APPROVED** [review]

File: dashboards-observability/public/components/application_analytics/__tests__/create.test.tsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    const setQuery = jest.fn();
+    const setFilters = jest.fn();
+    const setStartTime = jest.fn();
+    const setEndTime = jest.fn();
+    const dslService = {
+      http: jest.fn(),
+      fetch: jest.fn(),
+      fetchIndices: jest.fn(),
+      fetchFields: jest.fn()
+    } as unknown as DSLService;
```
Comment (mengweieric) [inline dashboards-observability/public/components/application_analytics/__tests__/create.test.tsx:28]: question: why both as unknown and DSLService?

File: dashboards-observability/public/components/application_analytics/components/application.tsx [hunk 2]
```diff
@@ -24,7 +24,7 @@ import DSLService from 'public/services/requests/dsl';
 import PPLService from 'public/services/requests/ppl';
 import SavedObjects from 'public/services/saved_objects/event_analytics/saved_objects';
 import TimestampUtils from 'public/services/timestamp/timestamp';
-import React, { ReactChild, useMemo, useState } from 'react';
+import React, { ReactChild, useState } from 'react';
```
Comment (mengweieric) [inline dashboards-observability/public/components/application_analytics/components/application.tsx:27]: Minor: maybe move react API and lodash importing to the top of the file.

File: dashboards-observability/public/components/application_analytics/components/config_components/log_config.tsx [hunk 3]
```diff
@@ -61,48 +61,48 @@ export const LogConfig = (props: LogConfigProps) => {
...
-            onClickAriaLabel={"pplLinkShowFlyout"}
-          >
-            PPL
-          </EuiBadge>
-        </EuiFlexItem>
-      </EuiFormRow>
-    </EuiAccordion>
-    {isModalVisible && modalLayout}
+        <EuiFormRow
+        label="PPL Base Query"
```
Comment (mengweieric) [inline dashboards-observability/public/components/application_analytics/components/config_components/log_config.tsx:82]: Minor: formatting

**Review by mengweieric — COMMENTED** [review]
Left minor comments, let's goooooo ship it!

**Pull request merged**